config/auth.go

```diff
@@ -293,3 +299,44 @@ func (o *AuthOIDCConfig) Validate() error {
 	}
 	return o.HTTPClientConfiguration.Validate()
 }
+
+type AuthKerberosAuthzConfig struct {
+	HTTPClientConfiguration `json:",inline" yaml:",inline"`
+	// Controls whether the authorization flow is enabled. If set to false
+	// all authenticated users are allowed in the service.
+	Enable bool `json:"enable" yaml:"enable" default:"false"`
+}
+
+func (k *AuthKerberosAuthzConfig) Validate() error {
+	if k.Enable {
+		return k.HTTPClientConfiguration.Validate()
+	}
+	return nil
+}
+
+type AuthKerberosClientConfig struct {
+	// Keytab is the path to the kerberos keytab
+	Keytab string `json:"keytab" yaml:"keytab" default:"/etc/krb5.keytab"`
+	// EnforceUsername specifies whether to check that the username of the
+	// authenticated user matches the SSH username entered. ...
+	EnforceUsername bool `json:"enforceUsername" yaml:"enforceUsername" default:"true"`
+	// CredentialCachePath is the path in which the kerberos credentials
+	// will be written inside the user containers. For kerberos to work
+	// correctly this should match the contents of the KRB5CC environment
+	// variable.
+	CredentialCachePath string `json:"credentialCachePath" yaml:"credentialCachePath" default:"/tmp/krb5cc"`
+	// AllowPassword controls whether kerberos-based password
+	// authentication should be allowed. If set to false only GSSAPI
+	// authentication will be permitted
+	AllowPassword bool `json:"allowPassword" yaml:"allowPassword" default:"true"`
+	// ConfigPath is the path of the kerberos configuration file. This is only used for password authentication.
+	ConfigPath string `json:"configPath" yaml:"configPath" default:"/etc/krb5.conf"`
```

Comment:
I would consider setting this to `/etc/containerssh/krb5.conf` so the container version automatically picks it up from the config directory.